source/CCustomOpcodeSystem.cpp

```diff
@@ -1623,45 +1617,71 @@ namespace CLEO {
 
 		*thread >> label >> nParams;
 
-		static SCRIPT_VAR arguments[64];
-		SCRIPT_VAR* arguments_end = &arguments[nParams];
+		ScmFunction* scmFunc = new ScmFunction(thread);
+		
+		static SCRIPT_VAR arguments[32];
+		SCRIPT_VAR* locals = thread->IsMission() ? missionLocals : thread->GetVarPtr();
+		SCRIPT_VAR* localsEnd = locals + 32;
+		SCRIPT_VAR* storedLocals = scmFunc->savedTls;
 
-		if (nParams)
+		// collect arguments
+		for (DWORD i = 0; i < min(nParams, 32); i++)
 		{
-			for (SCRIPT_VAR *arg = arguments; arg != arguments_end; ++arg)
+			SCRIPT_VAR* arg = arguments + i;
+				
+			switch (*thread->GetBytePointer())
 			{
-				switch (*thread->GetBytePointer())
+			case DT_FLOAT:
+			case DT_DWORD:
+			case DT_WORD:
+			case DT_BYTE:
+			case DT_VAR:
+			case DT_LVAR:
+			case DT_VAR_ARRAY:
+			case DT_LVAR_ARRAY:
+				*thread >> arg->dwParam;
+				break;
+
+			case DT_VAR_STRING:
+			case DT_LVAR_STRING:
+			case DT_VAR_TEXTLABEL:
+			case DT_LVAR_TEXTLABEL:
+				arg->pParam = GetScriptParamPointer(thread);
+				if (arg->pParam >= locals && arg->pParam < localsEnd) // correct scoped variable's pointer
 				{
-				case DT_FLOAT:
-				case DT_DWORD:
-				case DT_WORD:
-				case DT_BYTE:
-				case DT_VAR:
-				case DT_LVAR:
-				case DT_VAR_ARRAY:
-				case DT_LVAR_ARRAY:
-					*thread >> arg->dwParam;
-					break;
-				case DT_VAR_STRING:
-				case DT_LVAR_STRING:
-				case DT_VAR_TEXTLABEL:
-				case DT_LVAR_TEXTLABEL:
-					arg->pParam = GetScriptParamPointer(thread);
-					break;
-				case DT_STRING:
-				case DT_TEXTLABEL:
-					arg->pcParam = readString(thread);
+					arg->dwParam -= (DWORD)locals;
+					arg->dwParam += (DWORD)storedLocals;
 				}

```

Comment:
@MiranDMC can you explain this condition?

Reply:
I see, thanks. So this is essentially to support nested cleo_calls, to not override existing scope.